src/main/java/fi/fmi/avi/archiver/config/AmqpConfig.java

```diff
@@ -42,38 +44,46 @@ public AmqpConfig(
     }
 
     private static <T> T createLazyForwardingProxy(final Class<T> type, final Supplier<T> factory) {
-        final com.google.common.base.Supplier<T> delegate = Suppliers.memoize(factory::get);
+        final Supplier<T> delegate = Suppliers.memoize(factory::get);
         return type.cast(Proxy.newProxyInstance(
                 type.getClassLoader(),
                 new Class[]{type},
                 (proxy, method, args) -> method.invoke(delegate.get(), args)));
     }
 
+    private static void log(final Resource.Context context) {
+        switch (context.currentState()) {
+            case OPENING -> LOGGER.info("RabbitMQ connecting...");
+            case OPEN -> LOGGER.info("RabbitMQ connection established");
+            case RECOVERING -> LOGGER.info("RabbitMQ connection recovering...");
+            case CLOSING -> LOGGER.info("RabbitMQ connection closing...");
+            case CLOSED -> {
+                if (context.failureCause() != null) {
+                    LOGGER.error("RabbitMQ connection lost", context.failureCause());
+                } else {
+                    LOGGER.info("RabbitMQ connection closed");
+                }
+            }
+            default -> LOGGER.error("RabbitMQ connection in unknown state");
+        }
+    }
+
     @Bean
     public Environment amqpEnvironment() {
         return new AmqpEnvironmentBuilder().build();
     }
 
     @Bean
-    public Connection amqpConnection(final Environment environment) {
+    public Connection amqpConnection(final Environment environment,
```

Comment:
I just discovered that in this repo we have a convention of bean definitions and config class constructors being package-private instead of public (missed that in prior review, sorry!). Once cleaning up this config class, this convention could be applied here as well.